tpmutil/structures.go

```diff
@@ -52,6 +53,15 @@ func (b *U16Bytes) TPMUnmarshal(in io.Reader) error {
 		return err
 	}
 	size := int(tmpSize)
+
+	// If the reader is a bytes.Buffer, we check there are enough unread
+	// bytes to contain the data section indicated by the size prefix.
+	if bb, ok := in.(*bytes.Buffer); ok {
```

Comment:
Is this check necessary?
If remaining bytes in bytes.Buffer are fewer than size, in.Read below will fail with the same error

Reply:
I can get rid of this, I just thought it would be more correct for us to check the buffer before allocating memory. This isn't technically necessary if we have the 1mb check further on.